File: 20151007/src/beginner_tutorials/src/keyop.cpp
#include"ros/ros.h"
#include"geometry_msgs/Twist.h"

int main(int argc, char** argv)
{
  ros::init(argc, argv, "keyop");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<geometry_msgs::Twist>("mobile_base/commands/velocity", 1000);
  
  geometry_msgs::Twist twist;
  twist.linear.x = 0.5;
  twist.linear.y = 0;
  twist.linear.z = 0;
  twist.angular.x = 0.0;
  twist.angular.y = 0.0;
  twist.angular.z = 0.0;
  pub.publish(twist);
  ros::spinOnce();
  
  return 0;
}

File: 20151007/src/beginner_tutorials/scripts/keyop.py
#!/usr/bin/env python

import roslib
import rospy
import sys
from geometry_msgs.msg import Twist

def set_vel(x,y,t):
    pub = rospy.Publisher('mobile_base/commands/velocity', Twist)
    rospy.init_node('MYkeyop')
    set_vel = Twist()
    set_vel.linear.x = x
    set_vel.linear.y = y
    set_vel.angular.z = t
    while(1):
        pub.publish(set_vel)
        rospy.sleep(0.1)
if __name__ == '__main__':
    params = map(lambda x: float(x), sys.argv[1:])
    set_vel(params[0],params[1],params[2])
